pkg/network/ebpf/c/protocols/events.h

```diff
@@ -26,29 +26,36 @@
     static __always_inline void name##_flush_batch(struct pt_regs *ctx) {               \
         u32 zero = 0;                                                                   \
         batch_state_t *batch_state = bpf_map_lookup_elem(&name##_batch_state, &zero);   \
-        if (!batch_state || batch_state->idx_to_flush == batch_state->idx) {            \
+        if (!batch_state) {                                                             \
             /* batch is not ready to be flushed */                                      \
             return;                                                                     \
         }                                                                               \
+        _Pragma( STR(unroll(BATCH_PAGES_PER_CPU)) )                                     \
+            for (int i = 0; i < BATCH_PAGES_PER_CPU; i++) {                             \
+                if (batch_state->idx_to_flush == batch_state->idx) return;              \
```

Comment:
this would never be true (or only once at start) as idx_to_flush is always increasing (line 57)

line 57 should be
`batch_state->idx_to_flush = (batch_state->idx_to_flush++) % BATCH_PAGES_PER_CPU;`

Reply:
I'm not sure if I follow you. Both `batch_state->idx_to_flush` and and `batch_state->idx` are monotonic.
`batch->idx_to_flush` is _always_ less or equal to `batch->idx` which represents the active batch.
I haven't changed anything regarding these numbers or conditionals in this PR
Please take a look at
https://github.com/DataDog/datadog-agent/blob/907a713d5de73b2afaf73a40ad368747342ea09e/pkg/network/ebpf/c/protocols/events-types.h#L9-L20

Reply:
you right we don't need to bound the idx_to_flush as it's monotonic